src/memory/currency_manager.rs

```diff
@@ -0,0 +1,62 @@
+use crate::memory::memory_context::MemoryContext;
+use crate::memory::{MemoryManager, MemoryManagerUpdate};
+use crate::state::StateContext;
+use log::info;
+use memory::memory_manager::il2cpp::UnityMemoryManager;
+use memory::process::MemoryError;
+
+impl Default for MemoryManager<CurrencyManagerData> {
```

Comment:
I'm always a bit jarred when the `impl` comes before the actual `struct`, though I appreciate Rust for allowing it.

Reply:
Happy to make a cleanup ticket and knock this out tom - I agree as a preference but i dont pay attention